### PR TITLE
Resolve rename requests to `null` in Java files for qute-ls

### LINKED ISSUE #700: JavaFileTextDocumentService doesn't gracefully handle lack of rename support
Try to rename some element in a Java file when the Quarkus extension is started.

We retrieve the `TextDocumentService` and call rename on it. However `JavaFileTextDocumentService` doesn't support rename, and will throw an `UnsupportedOperationException`. The rename will still succeed.

https://github.com/redhat-developer/quarkus-ls/blob/635bedf3bc738108f21c228af27dd6d8d22fd8b8/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/QuteTextDocumentService.java#L233-L235

```
[Error - 12:56:35 p.m.] Request textDocument/rename failed.
  Message: Internal error.
  Code: -32603 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at com.redhat.qute.ls.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:148)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 12 more
Caused by: java.lang.UnsupportedOperationException
	at org.eclipse.lsp4j.services.TextDocumentService.rename(TextDocumentService.java:337)
	at com.redhat.qute.ls.QuteTextDocumentService.rename(QuteTextDocumentService.java:235)
	... 17 more
```

### PR DESCRIPTION
**Pull request by datho7561**
Closes #700

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/AbstractTextDocumentService.java [hunk 1]
```diff
@@ -167,6 +169,11 @@ public CompletableFuture<List<InlayHint>> inlayHint(InlayHintParams params) {
 		return CompletableFuture.completedFuture(null);
 	}
 
+	@Override
+	public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {
+		return CompletableFuture.completedFuture(null);
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/AbstractTextDocumentService.java:174]: If you wish to use this stragegy which implement method in AbstractTextDocumentService.java, please do that for other methods (like inlayHint).

Comment (datho7561) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/AbstractTextDocumentService.java:174]: I think it's already implemented this way

Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/AbstractTextDocumentService.java:174]: Good catch, it is perfect.

**Comment by angelozerr** [comment]
Thanks @datho7561 !

**Pull request merged**